src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -890,25 +889,80 @@ class SelectClauseModifierSegment(BaseSegment):
     )
 
 
-@snowflake_dialect.segment()
-class AlterTableColumnStatementSegment(BaseSegment):
-    """An `ALTER TABLE .. ALTER COLUMN` statement.
-
-    https://docs.snowflake.com/en/sql-reference/sql/alter-table-column.html
+@snowflake_dialect.segment(replace=True)
+class AlterTableStatementSegment(BaseSegment):
+    """An `ALTER TABLE` statement.
 
+    https://docs.snowflake.com/en/sql-reference/sql/alter-table.html
```

Comment:
Let's move your order comment here to avoid duplicating the URL:

```suggestion
    https://docs.snowflake.com/en/sql-reference/sql/alter-table.html
    If possible, please keep the order below the same as Snowflake's doc:
```